webapp/src/reducers/index.ts

```diff
@@ -70,7 +72,7 @@ function clientId(state = '', action) {
     }
 }
 
-function reviewDetails(state = [], action) {
+function reviewDetails(state = [], action: {type: string, data: Item}) {
```

Comment:
Also the reducers can be typed more explicitly like this:

```suggestion
import {Reducer} from 'redux';

const reviewDetails = Reducer<Item | null, {type: string, data: Item}>(state = null, action) {
```

I'm curious what you think about this syntax. Since these are abstractions around Redux, I think it makes sense to use their types to declare the reducers. I'll defer to you on this though.